# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/WindowGlobalParent.cpp
# Commit: acccc0eb50c9
# Full Hash: acccc0eb50c9a1111b4f152c608de7a230877385
# Author: Tom Ritter <tom@mozilla.com>
# Date: 2024-03-20 04:48:34
# Regressor Bug: 1884732
# File Overlap Count: 3
# Description:
#   Bug 1884732: Log JSActor messages with MMPrinter r=mccr8
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D204677
# ==============================================================================

diff -r 1957e76267e3 -r acccc0eb50c9 dom/ipc/WindowGlobalParent.cpp
--- a/dom/ipc/WindowGlobalParent.cpp	Tue Mar 19 17:46:39 2024 +0000
+++ b/dom/ipc/WindowGlobalParent.cpp	Tue Mar 19 18:58:16 2024 +0000
@@ -39,6 +39,7 @@
 #include "mozilla/Telemetry.h"
 #include "mozilla/Variant.h"
 #include "mozilla/ipc/ProtocolUtils.h"
+#include "MMPrinter.h"
 #include "nsContentUtils.h"
 #include "nsDocShell.h"
 #include "nsDocShellLoadState.h"
@@ -570,6 +571,8 @@
     stack.emplace();
     stack->BorrowFromClonedMessageData(*aStack);
   }
+  MMPrinter::Print("WindowGlobalParent::RecvRawMessage", aMeta.actorName(),
+                   aMeta.messageName(), *aData);
   ReceiveRawMessage(aMeta, std::move(data), std::move(stack));
   return IPC_OK();
 }
